### PR TITLE
PBL-42770: Make rocky-lint check node version

### PR DESCRIPTION
**Pull request by martijnthe**
Tested by using `nvm` to install 4.3.2 (too old) and also 5.10.0 (minimum required).

```
$ bin/rocky-lint ...
Node 5.10.0 is required, but 4.3.2 is used.
```

### CODE REVIEW

File: lib/rocky-lint.js [hunk 1]
```diff
@@ -46,6 +46,16 @@ const WARN = makeLogFilter(0);
 const INFO = makeLogFilter(1);
 const DEBUG = makeLogFilter(2);
 
+(function checkNodeVersion() {
+  const semver = require('semver');
+  const nodeVersionString = process.versions.node;
+  const minimumVersionString = '5.10.0';  // because we're using fs.mkdtemp
```
Comment (sirlantis) [inline lib/rocky-lint.js]: It would be great if this was retrieved from `package.json`

File: lib/rocky-lint.js [hunk 2]
```diff
@@ -46,6 +46,16 @@ const WARN = makeLogFilter(0);
...
+  if (!semver.satisfies(nodeVersionString, '>=' + minimumVersionString)) {
+    WARN(`Node ${minimumVersionString} is required, but ${nodeVersionString} is used.`);
```
Comment (sirlantis) [inline lib/rocky-lint.js]: We don't have a CI setup which would have told you: this line violates the max-line-length linter rule.

**Review by mrcljx — APPROVED** [review]
@martijnthe I pushed a commit to fix the mentioned issues. Can you take a look at it?

**Comment by martijnthe** [comment]
Looks good, thanks!

**Comment by martijnthe** [comment]
@sirlantis @HBehrens OK to bump the version to 2.1.0?

**Comment by sirlantis** [comment]
I opted to bump it to `2.0.1` since the version requirement was specified in `package.json` before, it now only does an early runtime validation on it (instead of breaking later with a cryptic error message).

**Pull request merged**